src/plugins/intel_gpu/src/plugin/transformations/fc_horizontal_fusion.cpp

```diff
@@ -0,0 +1,206 @@
+        }
+
+        std::shared_ptr<ov::Node> fused_zps;
+        if (zp_nodes.size() > 0) {
+            // scalar zp
+            auto zp_shape = zp_nodes[0]->get_output_shape(0);
+            bool is_scalar = (ov::shape_size(zp_nodes[0]->get_output_shape(0)) == 1);
+            int32_t scalar_zp_val = 0;
+            if (is_scalar) {
+                if (auto zp_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(zp_nodes[0])) {
+                    scalar_zp_val = zp_const->cast_vector<int32_t>()[0];
+                } else if (auto zp_convert = std::dynamic_pointer_cast<ov::op::v0::Convert>(zp_nodes[0])) {
+                    auto zp_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(zp_convert->get_input_node_shared_ptr(0));
+                    scalar_zp_val = zp_const->cast_vector<int32_t>()[0];
+                }
+                fused_zps = zp_nodes[0];
+            }
+            if (is_scalar) {
+                for (size_t i = 1; i < zp_nodes.size(); ++i) {
+                    bool current_is_scalar = (ov::shape_size(zp_nodes[i]->get_output_shape(0)) == 1);
+                    if (!current_is_scalar)
+                        return false;
+                    // validate all zp values are same
+                    int32_t cur_zp_val = 0;
+                    if (auto zp_const = std::dynamic_pointer_cast<ov::op::v0::Constant>(zp_nodes[i])) {
+                        cur_zp_val = zp_const->cast_vector<int32_t>()[0];
+                    } else if (auto zp_convert = std::dynamic_pointer_cast<ov::op::v0::Convert>(zp_nodes[i])) {
+                        auto zp_const =
+                            std::dynamic_pointer_cast<ov::op::v0::Constant>(zp_convert->get_input_node_shared_ptr(0));
+                        cur_zp_val = zp_const->cast_vector<int32_t>()[0];
+                    } else {
+                        OPENVINO_ASSERT("Unsupported zp input node for FC horizontal fusion");
+                    }
+                    if (cur_zp_val != scalar_zp_val)
+                        return false;
+                }
+            } else {
+                auto zp_nodes_as_output_vector = ov::OutputVector{zp_nodes[0], zp_nodes[1], zp_nodes[2]};
+                fused_zps = std::make_shared<ov::op::v0::Concat>(zp_nodes_as_output_vector, 0);
+                fused_zps->set_friendly_name(zp_nodes[0]->get_friendly_name() + "_fused");
+            }
+        }
+        // Create new fc with merged weights, bias, scale, zp
+        std::shared_ptr<ov::Node> new_fc;
+        if (fused_zps)
+            new_fc = std::make_shared<op::FullyConnectedCompressed>(input_node, fused_weight, fused_bias, fused_scale, fused_zps);
+        else
+            new_fc = std::make_shared<op::FullyConnectedCompressed>(input_node, fused_weight, fused_bias, fused_scale);
+
+        auto new_fc_name = fc_nodes[0]->get_friendly_name() + "_fused";
+        new_fc->set_friendly_name(new_fc_name);
+        copy_runtime_info(fc_nodes[0], new_fc);
```

Comment:
I think making `fc_nodes` ino `NodeVector` and put the entire `fc_nodes` to `copy_runtime_info`

Reply:
Fixed